packages/create-docusaurus/src/index.ts

```diff
@@ -452,7 +457,33 @@ export default async function init(
     getSiteName(reqName, rootDir),
   ]);
   const dest = path.resolve(rootDir, siteName);
-  const source = await getSource(reqTemplate, templates, cliOptions);
+
+  let language: {javascript?: boolean; typescript?: boolean} = {};
+  if (!cliOptions.typescript && !cliOptions.javascript) {
+    const {language: selectedLanguage} = (await prompts(
+      {
+        type: 'select',
+        name: 'language',
+        message: 'What language you want to use?',
+        choices: [
+          {title: 'JavaScript', value: 'javascript'},
+          {title: 'TypeScript', value: 'typescript'},
+        ],
+      },
+      {
+        onCancel() {
+          logger.info`Falling back to language=${'javascript'}`;
+        },
+      },
+    )) as {language: keyof Languages};
+
+    language = {[selectedLanguage]: true};
+  }
+
+  const source = await getSource(reqTemplate, templates, {
+    ...cliOptions,
+    ...language,
+  });
```

Comment:
not super fan of this merge between language and cli options. I don't think we should alter cli options